src/js/jsx/shared/Datalist.jsx

```diff
@@ -56,6 +59,8 @@ define(function (require, exports, module) {
             useAutofill: React.PropTypes.bool,
             // If true, will not highlight input text on commit
             neverSelectAllInput: React.PropTypes.bool,
+            // If true, mouse over an item will automatically select it and trigger the onChange callback.
+            autoSelect: React.PropTypes.bool,
```

Comment:
By default, Datalist will 1) automatically select any item that is hovered or selected via up/down keys. With the `live` option turned off, Datalist will not fire change event for every item select event, but 2) it will still fire a change event when the dialog is closed (by click on an item, or press ESC). The `autoSelect` option allows you to turn off behavior 1 and 2, and It will fire a change event only when a menu item is clicked.